### PR TITLE
[fix] Allow package name to not start with "com.google"

### LINKED ISSUE #2826: [Java] cannot generate io.grafeas protos
We're trying to generate the io.grafeas v1 protos from: https://github.com/googleapis/googleapis/tree/master/grafeas

The Java Gapic generator does not allow clients that do not follow `com.google.cloud.*` as the package name. https://github.com/googleapis/gapic-generator/blob/d204c06777c38e351ef665c69f0f31fa0b61c3bf/src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java#L455

### PR DESCRIPTION
**Pull request by hzyi-google**
Fix #2826.

### CODE REVIEW

**Comment by hzyi-google** [comment]
Generated code using artman with `--local` flag: https://github.com/hzyi-google/sample-staging/tree/master/20190617/gapic-google-cloud-grafeas-v1

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java [hunk 1]
```diff
@@ -438,24 +438,33 @@ public String getAndSaveTypeName(ImportTypeTable typeTable, TypeModel type) {
...
 
   public static String getExamplePackageName(String packageName) {
-    checkArgument(
-        packageName.startsWith("com.google."),
-        "We currently only support packages beginning with 'com.google'");
+    if (!packageName.startsWith("com.google.")) {
+      return packageName + ".examples";
+    }
+
+    // specical treatment for cloud clients because we anticipate them
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: sp: "special"

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java [hunk 2]
```diff
@@ -438,24 +438,33 @@ public String getAndSaveTypeName(ImportTypeTable typeTable, TypeModel type) {
   /**
    * Returns the package name of standalone samples.
    *
-   * <p>Currently we assume that package names always start with "com.google.". For example, if
-   * package name is "com.google.foo", the sample package name returned by this method will be
+   * <p>Most of the time package names start with "com.google.". For example, if package name is
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: s/if package name/if the package name/

Comment (vchudnov-g) [reply src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: In line 443 as well.

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java (hunk 1 above)
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: "anticipate they will be"

**Pull request merged**